nile-backend/src/main/java/com/tp/Nile/controllers/ReviewController.java

```diff
@@ -1,16 +1,87 @@
 package com.tp.Nile.controllers;
 
+import com.tp.Nile.exceptions.*;
 import com.tp.Nile.services.ReviewService;
+import com.tp.Nile.services.ReviewServiceImpl;
+import com.tp.Nile.models.Review;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
-@RequestMapping("/api")
+@RequestMapping("/api/reviews")
 @RestController
 @CrossOrigin(origins = "http://localhost:3000")
 public class ReviewController {
+
     @Autowired
-    ReviewService service;
+    ReviewServiceImpl service;
+
+    @PostMapping
+    public ResponseEntity addReview(@RequestBody Review review) {
+        try {
+            return ResponseEntity.ok(service.addReview(review));
+        }
+        catch (NullReviewIdException | InvalidReviewIdException | NullReviewAttributeException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+    }
+
+    @GetMapping
+    public ResponseEntity getReviews() {
+        try {
+            return ResponseEntity.ok(service.getAllReviews());
+        }
+        catch (InvalidReviewIdException | NullReviewIdException | NullReviewAttributeException e) {
+            return ResponseEntity.badRequest().body(e.getMessage());
+        }
+
+    }
+
+    @GetMapping("/{reviewId}")
```

Comment:
"/{reviewId}" and "/{userId}" will match the same pattern. Its probably better to use `@RequestParam`s for these anyway. But this is good.